dateformatterutil/src/main/java/com/example/dateformatter/DateFormatterUtil.kt

```diff
@@ -0,0 +1,182 @@
+        return date.toString(format)
+    }
+
+    /**
+     * @usage this function is used to convert LocalDateTime object to hijri date String
+     * @param localDate A LocalDate containing the date object
+     * @param isIslamic A Boolean confirm that the date object
+     * @return A Date that represents the localDate is islamic or normal date
+     */
+    fun convertFromIslamic(localDate: LocalDate, isIslamic: Boolean): Date? {
+        return if (isIslamic) {
+            val iso: Chronology = ISOChronology.getInstanceUTC()
+            val dtIso = localDate.toDateTimeAtStartOfDay(DateTimeZone.UTC)
+                .withChronology(iso).toLocalDate()
+            println("hijri-to-gregorian: $localDate=>$dtIso")
+            dtIso.toDate()
+        } else {
+            localDate.toDate()
+        }
+    }
+    /**
+     * @usage this function is used to convert date to hours and minutes based on UTC time zone
+     * @param date A String containing the date
+     * @return A String that represents the hours and minutes
+     */
+    fun convertDateToHoursMinUTC(dateString: String): String {
+        val format = SimpleDateFormat("HH:mm")
+        format.timeZone = TimeZone.getTimeZone("UTC")
+        val sdf = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss")
+        var d: Date? = null
+        try {
+            d = sdf.parse(dateString)
+        } catch (e: ParseException) {
+            e.printStackTrace()
+        }
+        return format.format(d)
+    }
+    /**
+     * @usage this function is used to convert date string to days and months
+     * @param date A String containing the date
+     * @return A String that represents the days and months
+     */
+    fun convertDateToDateDaysMonthsUTC(date: String): String {
+        val format = SimpleDateFormat("dd MMMM")
+        format.timeZone = TimeZone.getTimeZone("UTC")
+        val sdf = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss")
+        var d: Date? = null
+        try {
+            d = sdf.parse(date)
+        } catch (e: ParseException) {
+            e.printStackTrace()
+        }
+        return format.format(d)
+    }
+    /**
+     * @usage this function is used to compare two date
+     * @param fromDate A String containing the start date
+     * @param toDate A String containing the end date
+     * @return A Boolean that indicates the end date is after start or not
+     */
+    fun compareDates(fromDate: String , toDate: String): Boolean {
+        return Date(toDate).after(Date(fromDate))
+    }
+    /**
+     * @usage this function is used to split hours and min from date based on am and pm
+     * @param date A String containing the date
+     * @return A String that contains hh:min am/pm
+     */
+    fun getDatFromPmAmHoursMin(date: String): String {
+        val h1 = date.split(":".toRegex()).toTypedArray()
+        val hour = h1[0]
+        val minute = h1[1]
+        return if (date.contains("PM")) "$hour:$minute PM" else "$hour:$minute AM"
+    }
```

Comment:
First: Use an appropriate function naming as the current function name doesn't represent the actual usage of the function.
Second: This function is not null safe (if user passes a dateString that doesn't have hours and minutes format, the function will fail and application will crash), so it needs to be refactored.